website/docs/r/plugin.html.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: "vault"
+page_title: "Vault: vault_plugin resource"
+sidebar_current: "docs-vault-plugin"
+description: |-
+  Manage external plugins registered in the plugin catalog.
+---
+
+# vault\_plugin
+
+Manages external plugins registered in the plugin catalog.
+
+~> **Important** All data provided in the resource configuration will be
+written in cleartext to state and plan files generated by Terraform, and
+will appear in the console output when Terraform runs. Protect these
+artifacts accordingly. See
+[the main provider documentation](../index.html)
+for more details.
+
+For more information on managing external plugins, please refer to the Vault
+[documentation](https://developer.hashicorp.com/vault/docs/plugins).
+
+## Example Usage
+
+```hcl
+resource "vault_plugin" "jwt" {
+  type    = "auth"
+  name    = "jwt"
+  command = "vault-plugin-auth-jwt"
+  version = "v0.17.0"
+  sha256  = "6bd0a803ed742aa3ce35e4fa23d2c8d550e6c1567bf63410cec489c28b68b0fc"
+  env     = [
+    "HTTP_PROXY=http://proxy.example.com:8080"
+  ]
+}
+
+resource "vault_auth_backend" "jwt_auth" {
+  type = vault_plugin.jwt.name
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `type` - (Required) Type of plugin; one of "auth", "secret", or "database".
```

Comment:
I believe the answer to this is no, but can plugins be registered and pinned only for specific namespaces? If yes, we can also add in the [common namespace parameter](https://github.com/hashicorp/terraform-provider-vault/blob/main/website/docs/r/aws_secret_backend_role.html.md?plain=1#L53-L56) that is available for all resources.

If the `namespace` parameter is not required and can not be set for these resources, we can avoid using `UpdateSchemaResource` in [these lines](https://github.com/hashicorp/terraform-provider-vault/pull/2159/files#diff-7e69de666c9006fa0a0fb8fe4e4f8c418acce7c555f69f8087d88d16b434f2c1R783) of `provider.go`. That helper method adds the `namespace` param to the resource schema, and if they are unusable I think it's better to remove them as to not confuse users :)

Reply:
Yeah pinned versions are strictly a cluster-wide concept, no namespace-specific functionality. And thanks, good spot! I hadn't realised from the name of `UpdateSchemaResource` that that's what it was doing.